File: js/stockPicker.js
exports.picker = function(prices) {
  // pointers
  let buy = 0;
  let greatest = 0; //greatest profit
  let indices = [0,0]; // track indices to return
  // loop through stocks
  for (let i=0; i<prices.length; i++) {
    if (prices[i] < prices[buy]) {
      buy = i
    } else if (prices[i] - prices[buy] > greatest) {
      greatest = prices[i] - prices[buy]
      indices = [buy, i]
    }
  }
  return indices

}

File: python/stock_picker.py
# Challenge - no nested for loop, accomplish with a pointer in one loop

def picker(prices):
    # Input - prices - list of numbers
    buy = 0 # pointer for the buy date
    indices = [0, 0] # tracker for current profit generation
    greatest = 0
    for i in range(len(prices)):
        if prices[i] < prices[buy]:
            buy = i
        elif prices[i] - prices[buy] > greatest:
            greatest = prices[i] - prices[buy]
            indices = [buy, i]

    return indices

